src/pages/dashboard/change-email.tsx

```diff
@@ -0,0 +1,110 @@
+import { Box } from '@chakra-ui/react'
+import { useMutation } from '@tanstack/react-query'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+import CustomLoading from '@/components/CustomLoading'
+import ChangeEmailModal from '@/components/profile/components/ChangeEmailModal'
+import SuccessModal from '@/components/profile/components/SuccessModal'
+import { changeEmailWithToken } from '@/utils/api_helper'
+import { handleApiError } from '@/utils/error_helper'
+import { useToastHelpers } from '@/utils/toasts'
+
+const ChangeEmailPage = () => {
+  const router = useRouter()
+  const { token, address } = router.query
```

Comment:
Why are we using the address from a query ? The user needs to be logged in to do this , yes ?